core/string/ustring.cpp

```diff
@@ -3239,6 +3239,71 @@ String String::erase(int p_pos, int p_chars) const {
 	return left(p_pos) + substr(p_pos + p_chars);
 }
 
+String String::remove_char(char32_t p_char) const {
+	if (p_char == 0 || is_empty()) {
+		return *this;
+	}
+
+	String new_string;
+
+	new_string.resize(size());
+
+	char32_t *new_ptrw = new_string.ptrw();
+	const char32_t *old_ptr = ptr();
+
+	int new_size = 0;
+
+	while (*old_ptr) {
+		if (*old_ptr != p_char) {
+			new_ptrw[new_size] = *old_ptr;
+			++new_size;
+		}
+		old_ptr++;
+	}
+
+	new_ptrw[new_size] = 0;
+
+	new_string.resize(new_size + 1);
+
+	return new_string;
+}
+
+String String::remove_chars(const Vector<char32_t> &p_chars) const {
+	if (p_chars.is_empty() || is_empty()) {
+		return *this;
+	}
+
+	String new_string;
+
+	new_string.resize(size());
+
+	char32_t *new_ptrw = new_string.ptrw();
+	const char32_t *old_ptr = ptr();
+
+	int new_size = 0;
+
+	while (*old_ptr) {
+		bool found = false;
+		for (const char32_t &chr : p_chars) {
```

Comment:
Could nominally copy the vector instead above and sort it and then use a binary search but I don't think it'll be worth it generally as it'll almost always be a case of a handful of characters so `O(n)` Vs `O(log N)` won't really matter much I'd say